### ISSUE TITLE
is $lookup required (in lieu of square brackets ala js) to dynamically select object properties?

### ISSUE THREAD

**Issue by darrencruse** [jsonata-js/jsonata#447]
i.e. object properties whose names aren't known until run time?

to clarify: in plain javascript if I have something like this:
```
const json = {
    "param1": "esField1.keyword",
    "param2": "esField2.keyword"
}
```
it's typical to do just json.param1 if u know the "param1" part ahead of time
but if if u have a variable param whose value is the "param1" then (idiomatically) u switch to json[param]

In trying to use JSONata to generate some elasticsearch query json I struggled where I was trying to map input parameter names to elasticsearch field names using an object to do the mapping

I did finally get it working using $lookup like this:
https://try.jsonata.org/Fo2CJx6fE

But I wasted time I guess because the JSONata square bracket predicate syntax looks so much like javascript's I assumed I should have been able to do something more like this?
https://try.jsonata.org/rh3hfTkl4

(in my mind what I was doing seemed simple mostly I just wanted to confirm there wasn't a simple solution that didn't use the $lookup)

**Comment by markmelville** [comment]
I recently asked @andrew-coleman about this, and he said "I'm running out of tokens." My question, Andrew, is: can a token be interpreted differently depending on context:
`$[$prop]`
Could the [] token filter the array if the input is an array, and map to the property if it's an object? Currently if the argument is a string it always evaluates true, and so passes the object through no matter the value.

**Comment by darrencruse** [comment]
Yeah if it could (like u say) apply a rule that if the input before the square brackets is an object and what's inside the predicate square brackets evaluates to a string then effectively do the $lookup(input, predicateString) it feels like that would help the usability a lot (for people used to javascript!!)
